src/Catel.MVVM/Catel.MVVM.Shared/Collections/SuspensionContext.cs

```diff
@@ -15,11 +16,22 @@ namespace Catel.Collections
     /// <typeparam name="T">Type of the elements contained by the suspending collection.</typeparam>
     internal class SuspensionContext<T>
     {
+        /// <summary>
+        /// The events generator registry.
+        /// </summary>
+        private static Dictionary<SuspensionMode, Func<SuspensionContext<T>, ICollection<NotifyRangedCollectionChangedEventArgs>>> _eventGeneratorsRegistry;
+
+        /// <summary>
+        /// Gets events generators registry.
+        /// </summary>
+        public static Dictionary<SuspensionMode, Func<SuspensionContext<T>, ICollection<NotifyRangedCollectionChangedEventArgs>>> EventsGeneratorsRegistry => _eventGeneratorsRegistry ?? (_eventGeneratorsRegistry = new Dictionary<SuspensionMode, Func<SuspensionContext<T>, ICollection<NotifyRangedCollectionChangedEventArgs>>>());
```

Comment:
Not just create the dict, fill it up with all values.

Perhaps one of these:
1. public static Lazy<Dictionary<...>> EventsGeneratorsRegistry { get; } = new Lazy<Dictionary<...>>(() => { return InitFunction(); });
2. public static Dictionary<...> EventsGeneratorsRegistry => _eventGeneratorsRegistry ?? (_eventGeneratorsRegistry = InitFunction());